### PR TITLE
Extlinux syslinux fixes

### PR DESCRIPTION
**Pull request by dorileo**
+ revamps the extlinux/syslinux implementation to unify the common code and remove duplication
+ fix set-kernel for legacy bios mode (namely extlinux/syslinux implementation)
+ fix list-kernels for legacy bios mode

fixes: #217 #211

### CODE REVIEW

File: src/bootloaders/syslinux-common.c [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        boot_device = get_legacy_boot_device((char *)prefix);
+
+        if (!boot_device) {
+                boot_device = get_boot_device();
+        }
+
+        CHECK_ERR_RET_VAL(!boot_device, false, "No boot partition found, you need to "
+                          "mark the boot partition with \"legacy_boot\" flag.");
+
+	CHECK_ERR_GOTO(!writer(ctx, prefix, boot_device), cleanup,
```
Comment (bryteise) [inline src/bootloaders/syslinux-common.c]: Looks like some white-space formatting issue here.

File: src/bootloaders/syslinux-common.c [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        bool is_gpt = false;
+        struct SyslinuxContext *ctx;
+
+        ctx = boot_manager_get_data((BootManager *)manager);
+
+        prefix = boot_manager_get_prefix((BootManager *)manager);
+        boot_device = get_parent_disk((char *)prefix);
+
+        mbr = open(boot_device, O_WRONLY);
+	CHECK_ERR_RET_VAL(mbr < 0, false, "Could not open boot device: %s", boot_device);
```
Comment (bryteise) [inline src/bootloaders/syslinux-common.c]: Another white-space format issue I think.

File: src/bootloaders/syslinux-common.c [hunk 3]
```diff
@@ -23,6 +27,42 @@
...
+
+        f = fopen(config_path, "r");
+        CHECK_ERR_RET_VAL(!f, NULL, "Could not open config file: %s", config_path);
+
+        while ((read = getline(&buf, &len, f)) != -1) {
+                char *tk;
+
+                tk = strstr(buf, lookup);
+                if (tk != NULL) {
+                        result = strndup(tk+lplen, strlen(tk)-(lplen+1));
```
Comment (bryteise) [inline src/bootloaders/syslinux-common.c]: If the DEFAULT line doesn't have a newline I think this loses a character. Maybe not a problem though in practice but I think it is safer to just use len from getline and do `result = strdup(tk+lplen, len - lplen)` and then check if the second to last character of result is a newline and convert it into a null if it is. Thoughts?

**Comment by bryteise** [comment]
Seems fine overall with probably nit picking review X(.

File: src/bootloaders/syslinux-common.c (hunk 3 above)
Comment (dorileo) [reply src/bootloaders/syslinux-common.c]: should work, can't think why not.

**Comment by dorileo** [comment]
Ok

File: src/bootloaders/syslinux-common.c (hunk 1 above)
Comment (dorileo) [reply src/bootloaders/syslinux-common.c]: done. thanks.

File: src/bootloaders/syslinux-common.c (hunk 3 above)
Comment (dorileo) [reply src/bootloaders/syslinux-common.c]: done. thanks.

File: src/bootloaders/syslinux-common.c (hunk 2 above)
Comment (dorileo) [reply src/bootloaders/syslinux-common.c]: done. thanks.

**Comment by dorileo** [comment]
Just pushed a new version fixing the reviewed/mentioned issues.

**Comment by bryteise** [comment]
Looks good, thanks! Nice cleanup =).

**Pull request merged**